src/main/webapp/app/overview/course-discussion/course-discussion.component.html

```diff
@@ -31,33 +31,47 @@
                         <span>{{ 'artemisApp.metis.overview.postContextFilter' | artemisTranslate }}</span>
                     </div>
                     <div class="col-auto p-0">
-                        <select
+                        <mat-select
                             formControlName="context"
                             id="filter-context"
                             class="form-select course-discussion-select"
-                            [compareWith]="compareContextFilterOptionFn"
-                            (change)="onSelectContext()"
+                            panelClass="mat-select-panel"
+                            (openedChange)="$event === false ? onSelectContext() : null"
                             name="context"
+                            placeholder="{{ course.title + ' ' + ('artemisApp.metis.overview.courseFilterExplanation' | artemisTranslate) }}"
```

Comment:
You could consider moving the title into the translation string (translation string is something like `{{ title }} (All posts)``